### ISSUE TITLE
Fix `&` being detected as a duplicate selector

### ISSUE THREAD

**Issue by JacobDB** [stylelint/stylelint#3199]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When `no-duplicate-selectors` is enabled, every instance of a bare `&` triggers a warning.

> Which rule, if any, is this issue related to?

`no-duplicate-selectors`

> What CSS is needed to reproduce this issue?

```css
/* ------------------------------------------------------------------------ *\
 * Widget
\* ------------------------------------------------------------------------ */

.widget {
    & {
        border: 1px solid darken(#FFFFFF, 10);
        margin: 0 0 25px;
    }
}

.widget_title {
    & {
        background: #000000;
        color: #FFFFFF;
        font-size: 18px;
        font-weight: 400;
        margin-bottom: 0;
        padding: 10px 15px;
    }
}

.widget_content {
    & {
        padding: 10px 15px;
    }

    > :last-child {
        margin-bottom: 0 !important;
    }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "no-duplicate-selectors": true
  }
}
```

> Which version of stylelint are you using?

`9.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Gulp-stylelint

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it is related to SCSS and nesting.

> What did you expect to happen?

No warnings to be flagged. The `&` on its own resolves to the parent selector, and thus shouldn't be counted as a duplicate selector. I use this in order to make the top level rules line up more neatly with child rules; it's redundant, but it makes code easier to read for me.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
src/assets/styles/module/widget/_widget.scss
  6:5  ✖  Unexpected duplicate selector ".widget", first used at line 7            no-duplicate-selectors
 13:5  ✖  Unexpected duplicate selector ".widget_title", first used at line 14     no-duplicate-selectors
 24:5  ✖  Unexpected duplicate selector ".widget_content", first used at line 25   no-duplicate-selectors
```

**Comment by JacobDB** [comment]
The `&` on its own resolves to the parent selector, and thus shouldn't be counted as a duplicate selector. I use this in order to make the top level rules line up more neatly with child rules; it's redundant, but it makes code easier to read for me.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

This issue is a duplicate of #2198. Rule works as expected. Explanation starts with [this comment](https://github.com/stylelint/stylelint/issues/2198#issuecomment-2742761060).
